### PR TITLE
feat: Added zip support for sftp

### PR DESCRIPTION
**Pull request by afthabvp**
## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] New Connector (Destination or Source Connector)
- [ ] Breaking change (fix or feature that would impact existing functionality)
- [ ] Styling change
- [ ] Documentation update
- [ ] Refactoring
- [ ] Chore

## Checklist:

- [x] Ensure a branch name is prefixed with `feature`, `bugfix`, `hotfix`, `release`, `style` or `chore` followed by `/` and branch name e.g `feature/add-salesforce-connector`
- [x] Added unit tests for the changes made (if required)
- [ ] Have you made sure the commit messages meets the guidelines?
- [ ] Added relevant screenshots for the changes
- [ ] Have you tested the changes on local/staging?
- [x] Added the new connector in rollout.rb
- [ ] Have you updated the version number of the gem?
- [ ] Have you ensured that your changes for new connector are documented in the [docs repo](https://github.com/Multiwoven/docs) or relevant documentation files?
- [ ] Have you updated the run time dependency in multiwoven-integrations.gemspec if new gems are added
- [ ] Have you made sure the code you have written follows the best practises to the best of your knowledge?

### CODE REVIEW

File: integrations/lib/multiwoven/integrations/protocol/protocol.rb [hunk 1]
```diff
@@ -26,6 +26,8 @@ module Types
     LogLevel = Types::String.enum("fatal", "error", "warn", "info", "debug", "trace")
     RequestRateLimitingUnit = Types::String.default("minute").enum("minute", "hour", "day")
     SchemaMode = Types::String.enum("schema", "schemaless")
+    FormatType = Types::String.enum("csv")
```
Comment (subintp) [inline integrations/lib/multiwoven/integrations/protocol/protocol.rb:29]: FormatType is a generic name. Format can be anything

File: integrations/spec/multiwoven/integrations/destination/sftp/client_spec.rb [hunk 2]
```diff
@@ -101,6 +117,15 @@
       expect(response.tracking.failed).to eq(0)
     end
 
+    it "successfully writes records with compressed" do
+      allow(client).to receive(:with_sftp_client).and_yield(mock_sftp_session)
```
Comment (subintp) [inline integrations/spec/multiwoven/integrations/destination/sftp/client_spec.rb:121]: assert call to write_compressed_data as zero

File: integrations/spec/multiwoven/integrations/destination/sftp/client_spec.rb [hunk 3]
```diff
@@ -185,14 +210,14 @@
...
+      expect(file_path).to match(%r{/multiwoven/test_\d{8}-\d{6}\.csv\z})
     end
-  end
 
-  def sync_config
-    Multiwoven::Integrations::Protocol::SyncConfig.from_json(
-      sync_config_json.to_json
-    )
+    it "generate zip file" do
+      file_path = client.send(:generate_file_path, sync_config_compressed_zip)
```
Comment (subintp) [inline integrations/spec/multiwoven/integrations/destination/sftp/client_spec.rb:219]: assert call to write_uncompressed_data as 0

File: integrations/lib/multiwoven/integrations/protocol/protocol.rb [hunk 4]
```diff
@@ -26,6 +26,8 @@ module Types
...
+    CompressionType = Types::String.enum("un_compressed", "zip")
```
Comment (subintp) [inline integrations/lib/multiwoven/integrations/protocol/protocol.rb:30]: Raise PR for protocol changes in the documentation

**Review by subintp — CHANGES_REQUESTED** [review]

File: integrations/lib/multiwoven/integrations/protocol/protocol.rb (hunk 1 above)
Comment (afthabvp) [reply integrations/lib/multiwoven/integrations/protocol/protocol.rb:29]: i will change to File format type

**Review by subintp — APPROVED** [review]

**Pull request merged**